live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/auth/AuthenticationService.java

```diff
@@ -0,0 +1,177 @@
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.live_ingester.adapters.auth;
+
+import static org.apache.hc.core5.http.HttpHeaders.AUTHORIZATION;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientProperties;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.client.authentication.OAuth2LoginAuthenticationToken;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.user.DefaultOAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.security.oauth2.core.user.OAuth2UserAuthority;
+import org.springframework.security.scheduling.DelegatingSecurityContextTaskScheduler;
+import org.springframework.stereotype.Service;
+
+import org.alfresco.hxi_connector.live_ingester.adapters.config.IntegrationProperties;
+import org.alfresco.hxi_connector.live_ingester.adapters.config.properties.Authorization;
+import org.alfresco.hxi_connector.live_ingester.domain.exception.LiveIngesterRuntimeException;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService
+{
+    private static final String CLIENT_REGISTRATION_ID = "hyland-experience-auth";
+    private static final String APP_NAME_ATTRIBUTE_KEY = "applicationName";
+    private static final String SERVICE_USER_ATTRIBUTE_KEY = "serviceUser";
+    private static final String ENVIRONMENT_KEY_ATTRIBUTE_KEY = "hxAiEnvironmentKey";
+    private static final String ENVIRONMENT_KEY_HEADER = "hxai-environment";
+    private static final int AUTH_SCHEDULE_DELAY_MINUTES = 55;
+    private static final int WAIT_FOR_PAUSE_TIME_MILLIS = 100;
```

Comment:
It's probably worth moving some of these values to the configuration file (it could be in another PR though).

Reply:
I was thinking about moving this delay to configuration file - on it right now